eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -210,67 +210,24 @@ jobs:
   - checkout: self
     submodules: recursive
 
-  # Mac OS specific requirements
-  - task: Bash@3
-    inputs:
-      targetType: inline
-      script: |
-        sudo xcode-select --switch /Applications/Xcode_11.3.1.app
-        echo "xcode path:"
-        sudo xcode-select --print-path
-
-        # Install gcc 9
-        brew install gcc@9
-        gcc --version
-
-        git clone https://github.com/Microsoft/vcpkg.git
-        cd vcpkg
-        git rev-parse --verify HEAD
-        git status
-
-        ./bootstrap-vcpkg.sh
-
-        # Validate that vcpkg bootstrap succeeded
-        ./vcpkg version
-        if [ $? -ne 0 ]
-        then
-          echo "./bootstrap-vcpkg.sh FAILED"
-          exit 1
-        fi
-
-        echo "##vso[task.prependpath]$(pwd)"
-        echo "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT;]$(pwd)"
-
-    # Execute only for Mac and if there is at least one dependency to be installed
-    condition: and(succeeded(), contains(variables['vm.image'], 'macOS'), not(eq(variables['vcpkg.deps'], '')))
-    displayName: vcpkg bootstrap
-
-
-  - script: |
-      vcpkg install $(vcpkg.deps)
-    displayName: vcpkg install dependencies
-    # Execute only if there is at least one dependency to be installed
-    condition: and(succeeded(), not(eq(variables['vcpkg.deps'], '')))
+  - template: /eng/pipelines/templates/steps/vcpkg.yml
+    parameters:
+      VmImageVariableName: OSVmImage
```

Comment:
I wouldn't expect we need to pass this down any more given we are using a well known variable for the OS.